src/main/java/pl/psnc/dei/service/BatchService.java

```diff
@@ -307,6 +324,80 @@ private String extractIdentifier(String key) {
 		return currentProject.getDatasets();
 	}
 
+	/**
+	 * Gets all of the records for specified Europeana dataset (by dataset id)
+	 * and upload them to the local database
+	 * @param request which contains processing settings
+	 * @return set of records for whole Europeana dataset (except excluded records or duplications)
+	 * @throws IllegalArgumentException when missing required params
+	 * @throws NotFoundException when cannot found data in local database
+	 */
+	public Set<Record> uploadDataset(UploadDatasetRequest request)
+			throws IllegalArgumentException, NotFoundException {
+		String projectName = request.getProjectName();
+		String datasetId = request.getDatasetId();
+		if (projectName == null || datasetId == null) {
+			throw new IllegalArgumentException();
+		}
+		Set<String> recordsIds = europeanaSearchService.getAllDatasetRecords(datasetId);
+		Set<String> excludedFiltered = filterExcluded(recordsIds, request.getExcludedRecords());
+		Set<String> numberLimited;
+		if (request.getLimit() > 0) {
+			int recordsToRetrieve = Math.min(excludedFiltered.size(), request.getLimit());
+			numberLimited = excludedFiltered.stream()
+					.limit(recordsToRetrieve)
+					.collect(Collectors.toSet());
+		} else {
+			numberLimited = excludedFiltered;
+		}
+		return uploadRecordsToProject(projectName, request.getDataset(), numberLimited);
+    }
+
+	/**
+	 * Creates import for all records from Europena dataset (by dataset id).
+	 * except excluded records or duplications
+	 * @param request which contains processing settings
+	 * @return created imports
+	 * @throws IllegalArgumentException when missing required params
+	 * @throws NotFoundException when cannot found data in local database
+	 */
+    public List<Import> createImportsFromDataset(CreateImportFromDatasetRequest request)
+			throws IllegalArgumentException, NotFoundException {
+		Set<Record> uploadedDataset = uploadDataset(request);
+		String projectId = uploadedDataset.iterator().next().getProject().getProjectId();
+		int importSize = request.getImportSize();
+		List<Set<Record>> splitRecords;
+		boolean splitImport = importSize > 0;
```

Comment:
Maybe we should also think of a min importSize? Imagine that the user wants to upload a dataset of 10000 records...and he sets import size to 1...

Reply:
Added limitation of import sizes:
- min is 10% of whole records to retrieve
- max is 1000